### PR TITLE
Added Save in diktat smoke tests

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* added Save in diktat smoke tests

### CODE REVIEW

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt [hunk 1]
```diff
@@ -43,6 +47,13 @@ class TestComparatorUnit(private val resourceFilePath: String,
             copyTestFile.absolutePath
         )
 
+        if (smoke) {
+            val actual: MutableList<String> = mutableListOf()
+            actual.addAll(actualResult.split("\n"))
+            actual.removeLast()
```
Comment (petertrr) [inline diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt:53]:
```suggestion
            val actual = actualResult.split("\n").dropLast(1)
```

File: diktat-rules/src/test/resources/test/smoke/save.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[general]
+execCmd="./ktlint -R diktat-1.0.0-rc.2.jar"
```
Comment (petertrr) [inline diktat-rules/src/test/resources/test/smoke/save.toml:2]: I thought we were planning to execute SAVE in CI too?

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt (hunk 1 above)
Comment (Cheshiriks) [reply diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt]: Fixed

File: diktat-rules/src/test/resources/test/smoke/save.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+tags = ["smokeTest"]
+description = "SmokeTest"
+suiteName = "SmokeTest"
+
+[fix]
```
Comment (petertrr) [inline diktat-rules/src/test/resources/test/smoke/save.toml:7]: Shouldn't it be `fix-and-warn`? We have some unfixed warnings in smoke tests

Comment (Cheshiriks) [reply diktat-rules/src/test/resources/test/smoke/save.toml]: Fixed

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt [hunk 4]
```diff
@@ -21,10 +21,14 @@ class TestComparatorUnit(private val resourceFilePath: String,
...
      * @param testFileStr
+     * @param smoke
      * @return true if transformed file equals expected result, false otherwise
      */
     @Suppress("FUNCTION_BOOLEAN_PREFIX")
-    fun compareFilesFromResources(expectedResult: String, testFileStr: String): Boolean {
+    fun compareFilesFromResources(
+        expectedResult: String,
+        testFileStr: String,
+        smoke: Boolean = false): Boolean {
```
Comment (petertrr) [inline diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt:31]:
```suggestion
        trimLastEmptyLine: Boolean = false): Boolean {
```
To make it look like a more general option

**Review by petertrr — APPROVED** [review]

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt (hunk 4 above)
Comment (Cheshiriks) [reply diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt]: Fixed

File: diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/FileComparator.kt [hunk 5]
```diff
@@ -40,14 +40,22 @@ class FileComparator {
...
     fun compareFilesEqual(): Boolean {
         try {
             val expect = readFile(expectedResultFile.absolutePath)
             if (expect.isEmpty()) {
                 return false
             }
-            val patch = diff(expect, actualResultList)
+            val regex = (".*// ;warn:(\\d+):(\\d+): (.*)").toRegex()
+            val expectWithoutWarn = expect.filter { line ->
+                !line.contains(regex)
```
Comment (petertrr) [inline diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/FileComparator.kt:55]:
```suggestion
            val expectWithoutWarn = expect.filterNot { line ->
                line.contains(regex)
```

Comment (Cheshiriks) [reply diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/FileComparator.kt]: Fixed

**Pull request merged**